crates/test_utils/wasm/wasm_workspace/migrate_new/src/coordinator.rs

```diff
@@ -0,0 +1,84 @@
+use crate::integrity::*;
+use hdk::prelude::*;
+
+#[derive(Debug, serde::Serialize, serde::Deserialize, SerializedBytes)]
+struct Properties {
+    pub prev_dna_hash: holo_hash::DnaHash,
+}
+
+#[hdk_extern]
+fn init() -> ExternResult<InitCallbackResult> {
+    let properties: Properties = dna_info()
+        .unwrap()
+        .modifiers
+        .properties
+        .try_into()
+        .map_err(|_| {
+            wasm_error!(WasmErrorInner::Guest(
+                "Could not deserialize properties".to_string()
+            ))
+        })?;
+    open_chain(properties.prev_dna_hash.clone())?;
```

Comment:
This is something we talked about the other day. We can't expect the prev DNA hash to exist in the DNA properties in real apps, so how is this supposed to come in? Seems we'd have to do the open_chain in a zome call which can receive that data from the client?

Reply:
I think we can expect it to be present if we require a DNA to declare its lineage. So not part of the properties but somewhere on the DNA manifest. I think working out which DNA in the lineage, that the new version supports migration from, was actually present should be the conductors job and available somehow to the init callback.